signaler/signaler.go

```diff
@@ -1,25 +1,88 @@
+// Package signaler provides cross-platform signal handling for graceful application shutdown.
+//
+// The package automatically handles platform-specific signal differences and provides
+// both a simple legacy API and enhanced testing capabilities.
+//
+// Basic usage:
+//
+//	sig := signaler.WaitForInterrupt()
+//	log.Printf("Received %v, shutting down...", sig)
+//
+// The package correctly excludes uncatchable signals (like SIGKILL on Unix systems)
+// and includes appropriate signals for each platform.
 package signaler
 
 import (
 	"os"
 	"os/signal"
+	"runtime"
 	"syscall"
 )
 
+// s is the global signal channel used by WaitForInterrupt
+// it receives OS signals as configured by getPlatformSignals
 var s = make(chan os.Signal, 1)
 
+// notifier handles actual signal registration with the OS
+// I can be replaced with a mock for testing
+var notifier SignalNotifier = &osSignalNotifier{}
+
 func init() {
-	sigs := []os.Signal{
+	sigs := getPlatformSignals()
+	notifier.Notify(s, sigs...)
+}
+
+// getPlatformSignals returns the appropriate signals to listen for on the current platform.
+// On Unix-like systems (Linux, macOS, BSD), sigkill is excluded because it cannot be caught
+// or ignored. On Windows, sigkill is included because it can be caught.
+func getPlatformSignals() []os.Signal {
+	signals := []os.Signal{
 		os.Interrupt,
-		os.Kill,
 		syscall.SIGTERM,
 		syscall.SIGABRT,
 	}
-	signal.Notify(s, sigs...)
+
+	// Add os.Kill only for windows
+	// os.Kill cannot be caught or ignored on Unix-based systems
+	if runtime.GOOS == "windows" {
+		signals = append(signals, os.Kill)
+	}
+	return signals
+}
+
+// SignalNotifier is an interface for signal notification
+type SignalNotifier interface {
+	Notify(c chan<- os.Signal, sig ...os.Signal)
+	Stop(c chan<- os.Signal)
+}
```

Comment:
we can remove this as we can inject directly into the global `s` to test ` WaitForInterrupt`. Then its just the getPlatform signals which can be tested by our github actions and a runtime.Goos check. This will collapse a lot of bloat as what your testing signals is done in the signal package itself.